src/ai/contexts.cpp

```diff
@@ -1163,7 +1163,6 @@ void readonly_context_impl::recalculate_move_maps() const
 					++it;
 				}
 			}
-		///@todo: shall possible moves be modified as well ?
```

Comment:
Hmm, I just (re)noticed that this (removing moves for passive leaders from `dstsrc` etc) is only done for the leader of the current side. I am still not convinced that it needs to be done for `possible_moves` also, but the explanation in the commit message is not quite right.

Reply:
Looked into this some more, and I think even more so now that `possible_moves` should not have moves of passive leaders removed. It's supposed to be the *possible* moves, after all.  Filtering whether these should be used should happen in the CAs, not here.

If anything, I think that `dstsrc` and `srcdst` should retain those moves also, not the other way around. But there's a lot of weird stuff deep in the AI code here. For example, allied villages are removed from `srcdst` etc. I really don't think it should be done in this way. That's a task for reworking the AI as a whole though.

Bottom line: **I suggest simply removing this TODO comment.**